monitoring/mongodb/dashboard.py

```diff
@@ -0,0 +1,1397 @@
+    """Create a MongoDB Stat panel with common settings"""
+    if targets is None and expr:
+        targets = [Target(expr=expr, legendFormat="__auto")]
+    elif targets is None:
+        targets = []
+
+    # Set default values if not provided in kwargs
+    defaults = {
+        "colorMode": "value",
+        "textMode": "auto",
+    }
+
+    # Update defaults with any provided kwargs
+    for key, default_value in defaults.items():
+        if key not in kwargs:
+            kwargs[key] = default_value
+
+    return Stat(
+        title=title,
+        dataSource=DATASOURCE,
+        format=unit,
+        thresholds=thresholds or THRESHOLDS["green_red"],
+        targets=targets,
+        **kwargs,
+    )
+
+
+def mongodb_timeseries(title, targets, unit="none", thresholds=None, **kwargs):
+    """Create a MongoDB TimeSeries panel with common settings"""
+    # Set default values if not provided in kwargs
+    defaults = {
+        "drawStyle": "line",
+        "lineInterpolation": "linear",
+        "lineWidth": 1,
+        "fillOpacity": 10,
+        "pointSize": 5,
+        "showPoints": "never",
+    }
+
+    # Update defaults with any provided kwargs
+    for key, default_value in defaults.items():
+        if key not in kwargs:
+            kwargs[key] = default_value
+
+    return TimeSeries(
+        title=title,
+        dataSource=DATASOURCE,
+        unit=unit,
+        thresholds=thresholds,
+        targets=targets,
+        **kwargs,
+    )
+
+
+def mongodb_state_timeline(
+    title, expr, description="", mappings=None, width=None, **kwargs
+):
+    """Create a MongoDB StateTimeline panel"""
+    # Set default values if not provided in kwargs
+    defaults = {
+        "fillOpacity": 100,
+        "lineWidth": 0,
+        "showValue": "auto",
+        "mergeValues": True,
+        "rowHeight": 0.9,
+        "colorMode": "palette-classic",
+    }
+
+    # Update defaults with any provided kwargs
+    for key, default_value in defaults.items():
+        if key not in kwargs:
+            kwargs[key] = default_value
+
+    panel = StateTimeline(
+        title=title,
+        description=description,
+        dataSource=DATASOURCE,
+        mappings=mappings,
+        targets=[Target(expr=expr, legendFormat="{{pod}}")],
+        **kwargs,
+    )
+    if width:
+        panel.gridPos = GridPos(h=7, w=width, x=0, y=0)
+    return panel
+
+
+# Overview Panels
+mongodb_services_state = mongodb_stat(
+    "MongoDB services state",
+    orientation="horizontal",
+    targets=[
+        Target(
+            expr=MongoDBMetrics.up(JOBS["configsvr"]),
+            legendFormat="Config Server up:",
+        ),
+        Target(
+            expr=MongoDBMetrics.up(JOBS["mongos"]),
+            legendFormat="MongoS up:",
+        ),
+        Target(
+            expr=MongoDBMetrics.up(JOBS["shards"]),
+            legendFormat="Shard Server up:",
+        ),
```

Comment:
using python generator helps keep the code safe and simple in such case:
```suggestion
        Target(
            expr=MongoDBMetrics.up(JOBS[job]),
            legendFormat=f`{title} up:`,
        ) for job, title in {
            "configsvr": "Config Server",
            "mongos": "MongoS",
            "shards": "Shard Server", 
        } 
```

(and this may be further improved by adding the "title" in JOBS as well)

Reply:
this approach can be used in quite a few places in here to avoid redundancy and make code more expressive